src/Functions/FunctionSqid.cpp

```diff
@@ -0,0 +1,97 @@
+#include "config.h"
+
+#ifdef ENABLE_SQIDS
+
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Interpreters/Context.h>
+
+#include <sqids/sqids.hpp>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int SUPPORT_IS_DISABLED;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+// sqid(number1, ...)
+class FunctionSqid : public IFunction
+{
+public:
+    static constexpr auto name = "sqid";
+
+    String getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 0; }
+    bool isVariadic() const override { return true; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    static FunctionPtr create(ContextPtr context)
+    {
+        if (!context->getSettingsRef().allow_experimental_hash_functions)
```

Comment:
But it isn't a hashing function.

Reply:
Well, it transforms an (integer) input into a hash-like output ...

I think the only reason why the code checks `allow_experimental_hash_functions` is because the predecessor hashid function did so too (it was removed with #52449). I guess we don't check that setting at all. If there is a problem, the external code is simple enough to fix it.